runtime/js-compute-runtime/builtins/backend.cpp

```diff
@@ -806,41 +806,153 @@ JSString *Backend::name(JSContext *cx, JSObject *self) {
 }
 
 bool Backend::toString(JSContext *cx, unsigned argc, JS::Value *vp) {
-  METHOD_HEADER(0)
+  METHOD_HEADER(0);
   JS::RootedString name(cx, JS::GetReservedSlot(self, Backend::Slots::Name).toString());
   args.rval().setString(name);
   return true;
 }
 
-const JSFunctionSpec Backend::static_methods[] = {
-    JS_FS_END,
-};
-
-const JSPropertySpec Backend::static_properties[] = {
-    JS_PS_END,
-};
-
-const JSFunctionSpec Backend::methods[] = {JS_FN("toString", toString, 0, JSPROP_ENUMERATE),
-                                           JS_FS_END};
-
-const JSPropertySpec Backend::properties[] = {JS_PS_END};
-
-bool Backend::set_name(JSContext *cx, JSObject *backend, JS::HandleValue name_val) {
+namespace {
+JSString *parse_and_validate_name(JSContext *cx, JS::HandleValue name_val) {
   if (name_val.isNullOrUndefined()) {
     JS_ReportErrorNumberASCII(cx, GetErrorMessage, nullptr, JSMSG_BACKEND_NAME_NOT_SET);
-    return false;
+    return nullptr;
   }
   JS::RootedString name(cx, JS::ToString(cx, name_val));
   if (!name) {
-    return false;
+    return nullptr;
   }
   auto length = JS::GetStringLength(name);
   if (length > 254) {
     JS_ReportErrorNumberASCII(cx, GetErrorMessage, nullptr, JSMSG_BACKEND_NAME_TOO_LONG);
-    return false;
+    return nullptr;
   }
   if (length == 0) {
     JS_ReportErrorNumberASCII(cx, GetErrorMessage, nullptr, JSMSG_BACKEND_NAME_EMPTY);
+    return nullptr;
+  }
+  return name;
+}
+} // namespace
+
+bool Backend::exists(JSContext *cx, unsigned argc, JS::Value *vp) {
+  JS::CallArgs args = JS::CallArgsFromVp(argc, vp);
+  JS::RootedObject self(cx, &args.thisv().toObject());
+  if (!args.requireAtLeast(cx, "Backend.exists", 1)) {
+    return false;
+  }
+
+  JS::RootedString name(cx, parse_and_validate_name(cx, args.get(0)));
+  if (!name) {
+    return false;
+  }
+  auto nameChars = core::encode(cx, name);
+  std::string_view name_view = nameChars;
+  auto res = host_api::Backend::exists(name_view);
+  if (auto *err = res.to_err()) {
+    HANDLE_ERROR(cx, *err);
+    return false;
+  }
+  auto exists = res.unwrap();
+  args.rval().setBoolean(exists);
+  return true;
+}
+
+bool Backend::fromName(JSContext *cx, unsigned argc, JS::Value *vp) {
+  JS::CallArgs args = JS::CallArgsFromVp(argc, vp);
+  JS::RootedObject self(cx, &args.thisv().toObject());
+  if (!args.requireAtLeast(cx, "Backend.fromName", 1)) {
+    return false;
+  }
+
+  JS::RootedString name(cx, parse_and_validate_name(cx, args.get(0)));
+  if (!name) {
+    return false;
+  }
+  auto nameChars = core::encode(cx, name);
+  std::string_view name_view = nameChars;
+  auto res = host_api::Backend::exists(name_view);
```

Comment:
`host_api::HostString` will automatically convert to a `std::string_view`, so you can skip the intermediate definition :)
```suggestion
  auto res = host_api::Backend::exists(nameChars);
```